zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -102,6 +122,8 @@ private[http] object FormState {
     def reset(): Unit = {
       tree0.clear()
       buffer.clear()
+      bufferSize = 0
+      boundaryMatches.map(_ => false)
```

Comment:
What's usually the size of the array? Might be more performant to just make it a `var` and replace the whole thing?

Reply:
see b338c6e347ace6ea3cb0bd5d918e3ce4dcdfa16b